worlds/cv64/client.py

```diff
@@ -0,0 +1,168 @@
+from typing import TYPE_CHECKING, Optional, Set
+from .locations import base_id
+from .text import cv64_text_wrap, cv64_string_to_bytes
+
+from NetUtils import ClientStatus
+import worlds._bizhawk as bizhawk
+from worlds._bizhawk.client import BizHawkClient
+
+if TYPE_CHECKING:
+    from worlds._bizhawk.context import BizHawkClientContext
+else:
+    BizHawkClientContext = object
+
+
+class Castlevania64Client(BizHawkClient):
+    game = "Castlevania 64"
+    system = "N64"
+    patch_suffix = ".apcv64"
+    self_induced_death = False
+    received_deathlinks = 0
+    death_link = False
+    local_checked_locations: Set[int]
+    rom_slot_name: Optional[str]
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.local_checked_locations = set()
+        self.rom_slot_name = None
+
+    async def validate_rom(self, ctx: BizHawkClientContext) -> bool:
+        from CommonClient import logger
+
+        try:
+            # Check if ROM is some version of Castlevania 64
+            game_name = ((await bizhawk.read(ctx.bizhawk_ctx, [(0x20, 0x14, "ROM")]))[0]).decode("ascii")
+            if game_name != "CASTLEVANIA         ":
+                return False
+            
+            # Check if we can read the slot name. Doing this here instead of set_auth as a protection against
+            # validating a ROM where there's no slot name to read.
+            try:
+                slot_name_bytes = (await bizhawk.read(ctx.bizhawk_ctx, [(0xBFBFE0, 0x10, "ROM")]))[0]
+                self.rom_slot_name = bytes([byte for byte in slot_name_bytes if byte != 0]).decode("utf-8")
```

Comment:
"Slot name" is perhaps an inadequate description of what you're actually storing here. Also just want to double check that filtering all null bytes is always fine and expected, since I know this line was copied from Emerald.

Reply:
Made this comment before I recognized that your `modify_multidata` doesn't seem to add any extra auth keys.